compiler/passes/checkParsed.cpp

```diff
@@ -514,3 +520,35 @@ checkExportedNames()
     names.put(name, true);
   }
 }
+
+static void warnUnstableUnions(AggregateType* at) {
+  if (fWarnUnstable && at->isUnion()) {
+    USR_WARN(at, "Unions are currently unstable and are expected to change in ways that will break their current uses.");
+  }
+}
+
+static void warnUnstableLeadingUnderscores() {
+  if (fWarnUnstable) {
+    forv_Vec(VarSymbol, var, gVarSymbols) {
+      if (var->name[0] == '_' &&
+          var->defPoint->getModule()->modTag == MOD_USER &&
+          !var->hasFlag(FLAG_TEMP)) {
+        USR_WARN(var->defPoint,
+                 "Symbol names with leading underscores (%s) are unstable.", var->name);
+      }
+    }
+    forv_Vec(FnSymbol, fn, gFnSymbols) {
+      if (fn->name[0] == '_' &&
+          fn->defPoint->getModule()->modTag == MOD_USER) {
+        USR_WARN(fn->defPoint,
+                 "Symbol names with leading underscores (%s) are unstable.", fn->name);
+      }
+    }
+    forv_Vec(ModuleSymbol, mod, gModuleSymbols) {
+      if (mod->name[0] == '_' && mod->modTag == MOD_USER) {
+        USR_WARN(mod->defPoint,
+                 "Symbol names with leading underscores (%s) are unstable.", mod->name);
+      }
+    }
```

Comment:
Is this because we don't have collectors that contain all Symbols?  Because I think we'd also want to error about enums and class/record definitions, and potential other definitions that I'm not thinking of right now.

Reply:
Thanks! I updated it to catch a wider swath of symbol types using a simpler check.